docs/assignment3.md

```diff
@@ -1 +1,577 @@
+
+<img src="https://i.postimg.cc/vHbZ16dK/Screen-Shot-2020-03-30-at-9-35-39-AM.png" alt="Image of Client Interface" style="zoom:30%;" />
+
+## State machine diagrams									
+Author(s): `Irene Garcia-Fortea Garcia, Florent Brunet de Rochebrune`
+
+<h5>State-Machine Diagram for Getting Inputs for Player</h5>
+
+> State machine diagram of Game (Diagram 5) **DESCRIPTIVE**
+
+[![State-Machine-Diagram-game-class.png](https://i.postimg.cc/cHtqYYT2/State-Machine-Diagram-game-class.png)](https://postimg.cc/1nSY18kr)
+
+The diagram above shows the State Machine Diagram of the event that is executing a command. After starting the process, the system waits for a user input. After the input has been received, the function *getCommand()* executes. 
+
+In the next step we move to the state that the command has been received. The command needs to be checked, whether it is a valid command that the system accepts in *validateCommand()*. 
+
+If the command is not valid, an error message is printed and the program returns to the idle state, waiting for a command. 
+
+If the command is valid, the command needs to be executed ("*Execute Command*"). There, changes will be made to the Item or Player (Described in the *Execute Command* figure). After the changes have been made, the system goes back to the idle state, waiting for a new command to process, unless the user types “quit”. Then, the game terminates. 
+
+The *Execute Command* state works as the following: After checking the command for validity, it will be matched matching it with each of the following existing commands: move, help, look, list, drop, take/get/pickup, give, score and text/deliver/broadcast. 
+
+If the command is move: the player will be moved into the wanted direction (input by the player).
+
+If the command is help: a help message will be printed to the user.
+
+If the command is look: the description of the *Area*, in that the *Player* is currently in, will be printed to the user.
+
+If the command is list: a list of available commands will be displayed to the user.
+
+If the command is drop: the player will drop the *Item* (input by the player) from the backpack into the current *Area*.
+
+If the command is take/get/pickup: the object from the current *Area* will be placed into the backpack (if the *Area* has it).
+
+If the command is give: the player will give the *Item* to the *Obstacle* in the current *Area*. E.g. wallet to guard.
+
+If the command is score: the score of the user will be printed.
+
+If the command is text/deliver/broadcast: the input message will be sent to every other *Player* currently in the server.
+
+<h5>State-Machine Diagram for Player Movement</h5>
+
+> Figure representing the state machine diagram of the class Map (Diagram 6) **DESCRIPTIVE**
+
+[![State-Machine-Diagram-map-class.png](https://i.postimg.cc/RCf5Z7Wn/State-Machine-Diagram-map-class.png)](https://postimg.cc/0r8HtJd8)
```

Comment:
Some of the internal states that only have activities could have been ommited, and the activity could have been included in the transition.